File: Lv.1/과일 장수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
#include <functional>
using namespace std;

int solution(int k, int m, vector<int> score) {
    int answer = 0;

    sort(score.begin(),score.end(),greater<int>()); // 내림차순 정렬
    for(int i=m-1;i<score.size();i+=m){
        answer+=score[i]*m;
    }

    return answer;
}

File: Java/LV1/과일 장수.java
import java.util.*;
class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        //3322111
        Integer[] sc = new Integer[score.length];
        
        for(int i=0;i<sc.length;i++){
            sc[i]=score[i];
        }
        
        Arrays.sort(sc,Collections.reverseOrder());
        int idx = m-1;
        while(idx<score.length){
            answer += sc[idx]*m;
            idx+=m;
        }
        return answer;
    }
}
//1112233
// 1: 2
// 2: 4
// 4: 6
// 112 222 444 444
// 3 6 12 12
